Makefile.toml

```
@@ -402,6 +413,44 @@ fi
 '''
 ]
 
+[tasks.check-shell]
+script = [
+'''
+rc=0
+
+# For bash first-party shell code
+if ! docker run --rm \
+  --network=none \
+  --user "$(id -u):$(id -g)" \
+  --security-opt label:disable \
+  -v "${BUILDSYS_TOOLS_DIR}":/tmp/tools \
+  "${BUILDSYS_SDK_IMAGE}" \
+  bash -c \
+    'flagged_scripts=0 && \
+    cd /tmp/tools && \
+    for shell_script in $(grep -l --directories=skip --regexp="^#!.*bash$" * ); do
+      if ! shellcheck \
+        --external-sources \
+        --source-path=SCRIPTDIR \
+        --format=gcc \
+        --severity=warning \
+        "${shell_script}"; then
+        ((++flagged_scripts))
+      fi
+    done && \
+    if [ ${flagged_scripts} -ne 0 ]; then
+      exit 1
+    fi'; then
```

nit: since I am in the "quote and brace everything" school

```suggestion
    if [ "${flagged_scripts}" -ne 0 ]; then
      exit 1
    fi'; then
```

---

I tend to do "quote and brace everything" UNLESS the variable value is an integer or a variable name is single character, but I'll go ahead and make this change since I have to make another revision already.